src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1714,6 +1714,33 @@ abstract class RefChecks extends Transform {
           checkImplicitViewOptionApply(tree.pos, fn, args)
           checkSensible(tree.pos, fn, args) // TODO: this should move to preEraseApply, as reasoning about runtime semantics makes more sense in the JVM type system
         }
+        def lintedBoolean = {
+          val sym = fn.symbol
+          (  sym.ne(null)
+          && sym.ne(NoSymbol)
+          && !sym.isJavaDefined
+          && !sym.isSetter
+          && (
+            sym.name match {
+              case nme.apply =>
+                val companion = sym.owner.companion
+                companion != SomeClass && companion != OptionClass && companion != ArrayClass && !isTupleSymbol(companion) && !isFunctionSymbol(sym.owner)
+              case nme.update =>
+                args.lengthCompare(1) != 0
+              case _ => true
+            }
+          ))
+        }
+        def innommable(s: Symbol) = s.deprecatedParamName match {
+          case Some(nme.NO_NAME) => true
+          case _ => false
+        }
+        if (settings.lintNamedBooleans && lintedBoolean)
+          foreach2(args, fn.symbol.paramss.head)((arg, param) => arg match {
+            case t @ Literal(Constant(_: Boolean)) if t.hasAttachment[UnnamedArg.type] && !innommable(param) && !isRepeated(param) =>
```

Comment:
Did you consider `&& param.tpe.typeSymbol == BooleanClass`? This would cover the special cases for Option / Array / etc above. Would it eliminate some useful warnings?

For example this PR adds `newLiteral(value = true)` above, which I think is not necessary.

Reply:
I added deprecatedName to a couple of methods such as `newLiteral` and `tpd.Constant`, which take a `value` which is just the value.

Reply:
It still seems to me that warning only if the paramter type is a boolean would be right. This will also come up in user code.